src/Microsoft.OData.Core/UriParser/ODataUriParserSettings.cs

```diff
@@ -138,6 +139,11 @@ public int MaximumExpansionCount
             }
         }
 
+        /// <summary>
+        /// True if url parser support parsing path with key as segment, otherwise false. The default is true.
+        /// </summary>
+        public bool EnableParsingKeyAsSegmentUrl { get; set; }

```

Comment:
Why `EnableParsingKeyAsSegmentUrl` and not `EnableParsingKeyAsSegment`?

Reply:
That's how the property was called in `ODataSimplifiedOptions`. I did not change the names of the properties or methods, just moved them to different classes. Do you think I should rename it?